### DISCUSSION TITLE
west flash error

### DISCUSSION THREAD

**Discussion by jk0010** [zephyrproject-rtos/zephyr#67557] [Q&A] [answered]
Nucleo STM32F401RE
zephyr-sdk-0.16.4

Hello all,
I am brand new to zephyr and still figuring stuff out. I am having some trouble flashing to my nucleo board. I am able to build a project just fine ` west build -p auto -b nucleo_f401re samples/basic/blinky`. The error I get when I run `west flash` is 

```
-- west flash: rebuilding
[1/1] cd /home/lab/zephyrproject/zephyr/build/zephyr && /usr/bin/python3 /home/lab/zephyrpro...d/check_init_priorities.py --elf-file=/home/lab/zephyrproject/zephyr/build/zephyr/zephyr.elf
-- west flash: using runner openocd
-- runners.openocd: Flashing file: /home/lab/zephyrproject/zephyr/build/zephyr/zephyr.hex
Open On-Chip Debugger 0.11.0+dev-00726-ge2b6f5655 (2023-11-15-11:07)
Licensed under GNU GPL v2
For bug reports, read
        http://openocd.org/doc/doxygen/bugs.html
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
Info : clock speed 2000 kHz
Error: open failed


FATAL ERROR: command exited with status 1: /home/lab/zephyr-sdk-0.16.4/sysroots/x86_64-pokysdk-linux/usr/bin/openocd -s /home/lab/zephyrproject/zephyr/boards/arm/nucleo_f401re/support -s /home/lab/zephyr-sdk-0.16.4/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts -f /home/lab/zephyrproject/zephyr/boards/arm/nucleo_f401re/support/openocd.cfg '-c init' '-c targets' -c 'reset init' -c 'flash write_image erase /home/lab/zephyrproject/zephyr/build/zephyr/zephyr.hex' -c 'reset run' -c shutdown

```
Any help or suggestions would be much appreciated.
Thank you

**Comment by Tom2096** [comment]
Hi! Did you end up fixing this issue? I am running into something similar.

**Reply by jk0010** [reply, marked as answer]
Yes I did. Before I started getting this error I installed jlink drivers to try to solve a different issue. After this my board was recognized as a JLink device in device manager. I believe this was the cause of my problem. First I'd check how your device is being recognized in device manager.

My soultion:
I installed the SEGGER STLinkReflash utility and followed the restore stlink instructions. After this my device was recognized as STLink in device manager (previously JLink). Running `lsusb` and the device was recognized as STLink. After this I was able to flash my nucleo board.

**Reply by Tom2096** [reply]
Thanks, reflashing back to STLink and updating the firmware did the trick.
